.github/workflows/build-check.yml

```diff
@@ -15,11 +15,11 @@ jobs:
     runs-on: ubuntu-latest
 
     steps:
-    - uses: actions/checkout@v2
-    - name: Set up JDK 8
+    - uses: actions/checkout@v3
+    - name: Set up JDK 11
       uses: actions/setup-java@v1
       with:
-        java-version: '1.8'
+        java-version: 11
```

Comment:
Let us move this to 17 directly.

Reply:
Done this with temurin, let me know if you want to do adopt. Also updated jacoco version to work with this to 0.8.8 (from 0.8.6)

Reply:
Done.